### PR TITLE
First version of Consul integration with Rudder

### PR DESCRIPTION
**Pull request by ikramBej**

### CODE REVIEW

File: consul/build.conf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# Unique identifier of the plugin
+plugin-name=consul
+# the full name is derived from rudder-plugin-name
+plugin-fullname=rudder-plugin-${plugin-name}
+
+# Human readable short/title descrption (used for one line text)
+plugin-title-description="""Plugin for Consul integration with Rudder"""
+
+# WEB, HTML description. 
+plugin-web-description=<p>Plugin for Consul integartion with Rudder</p>
```
Comment (amousset) [inline consul/build.conf:18]: integration

File: consul/build.conf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# Plugin version. It is build as follow: A.B-x.y(.z) with:
+# - A.B: Rudder major.minor
+# - x.y(.z): plugin major.minor.micro. Micro should be omitted. When omitted, z is assumed to be 0. 
+# For the build, we split the information between two properties, rudder branch and plugin version, 
+# which must be concaneted with "-" to build the plugin version. 
+plugin-branch=1.2-SNAPSHOT
```
Comment (amousset) [inline consul/build.conf:25]: 1.0-SNAPSHOT

File: consul/share/plugins/consul/20_cfe_basics/consul_lib.cf [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#####################################################################################
+# Copyright 2018 Normation SAS
```
Comment (amousset) [inline consul/share/plugins/consul/20_cfe_basics/consul_lib.cf:2]: 2020

File: consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf [hunk 4]
```diff
@@ -0,0 +1,74 @@
+#####################################################################################
+# Copyright 2018 Normation SAS
```
Comment (amousset) [inline consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf:2]: 2020

File: consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+bundle agent variable_from_consul(variable_prefix, variable_name, path)
+{
+  vars:
+      "old_class_prefix" string => canonify("variable_from_consul_${path}");
+      "promisers" slist => { @{this.callers_promisers}, cf_null }, policy => "ifdefined";
+      "class_prefix" string => canonify(join("_", "promisers"));
+      "args" slist => {"${variable_prefix}", "${variable_name}", "${path}"};
+
+    pass1::
+      "${variable_prefix}.${variable_name}" string => execresult('/usr/bin/curl -s -H "X-Consul-Token: ${consul.config[token]}" http://${consul.config[server_addr]}/v1/kv/${pa$
```
Comment (amousset) [inline consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf:44]: The end of this line seems broken (probably a copy/paste issue)

Comment (ikramBej) [reply consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf:44]: oups! j'ai pas fait attention

**Pull request merged**